internal/pkg/gateway/registry.go

```diff
@@ -348,7 +343,41 @@ func (reg *registry) registerChannel(channel string) error {
 			}
 		}
 	}
-	reg.channelsInitialized[channel] = true
+	reg.channelInitialized[channel] = true
 
 	return nil
 }
+
+func (reg *registry) updateConfig(channel string) ([]*endpointConfig, error) {
```

Comment:
Good catch Manish. I think that maybe there should be a critical section with sync.Once where you do a preliminary loading of the channel's config and this would ensure that you can never get to a state where the channel's orderers aren't populated outside the sync.Once block.

Something like this:
1. Check if **reg.channelOrderers.Load(channel)** is empty
2. If not, great.
3. Else, it's empty. Load from discovery.Config inside a sync.Once block where you check if it has already been populated. 

wdyt?

Reply:
Alternatively, change `updateConfig()` so that it only reads and returns the orderer config rather than store it, then when invoked from (1), follow it with LoadOrStore(), and when invoked from (2), follow it with Store().  That would ensure that in a race condition, the channelOrderers sync.Map would always end up with the config from (2).

Reply:
Yes, Both should work. I guess, the one that Andy suggested causes minimum changes in the current code.